src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -444,27 +442,31 @@ static int lease_parse_search_domains(const uint8_t *option, size_t len, char **
                                         return r;
 
                                 n += r;
-                                continue;
                         } else if ((c & 0xc0) == 0xc0) {
                                 /* Pointer */
 
                                 uint8_t d;
                                 uint16_t ptr;
+
+                                if (pos >= len)
+                                        return -EMSGSIZE;
+
                                 d = option[pos++];
                                 ptr = (uint16_t) (c & ~0xc0) << 8 | (uint16_t) d;
 
-                                /* Jumps are limited to a prior occurrence of a complete domain (labels ending with 0),
-                                 * according to RFC1035 4.1.4. */
+                                /* Jumps are limited to a "prior occurrence" (RFC-1035 4.1.4) */
                                 if (ptr >= jump_barrier)
                                         return -EBADMSG;
+                                jump_barrier = ptr;
 
-                                /* Return to where we were as soon as the referred domain is parsed.*/
-                                next_chunk = pos;
+                                /* Return to where we were as soon as the referred domain is parsed. */
```

Comment:
This comment was confusing to me. This is the point where we just _save_ the position to return to. _Returning_ to it happens in lines 481-482.